instrumentation/elasticsearch/elasticsearch-api-client-8.0/javaagent/src/test/java/ElasticsearchClient8Test.java

```diff
@@ -0,0 +1,254 @@
+                    builder
+                        .setConnectTimeout(Integer.MAX_VALUE)
+                        .setSocketTimeout(Integer.MAX_VALUE))
+            .build();
+
+    ElasticsearchTransport transport =
+        new RestClientTransport(restClient, new JacksonJsonpMapper());
+    client = new ElasticsearchClient(transport);
+    asyncClient = new ElasticsearchAsyncClient(transport);
+  }
+
+  @AfterAll
+  static void cleanUp() {
+    elasticsearch.stop();
+  }
+
+  private static String userAgent() {
+    return "elastic-java/"
+        + RestClientBuilder.VERSION
+        + " (Java/"
+        + System.getProperty("java.version")
+        + ")";
+  }
+
+  @Test
+  // ignore deprecation interface
+  public void elasticsearchStatus() throws IOException {
+    InfoResponse response = client.info();
+    Assertions.assertEquals(response.version().number(), "7.17.2");
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("info")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasNoParent()
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.DB_SYSTEM, "elasticsearch"),
+                            equalTo(SemanticAttributes.DB_OPERATION, "info"),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                            equalTo(SemanticAttributes.HTTP_URL, httpHost.toURI() + "/")),
+                span ->
+                    span.hasName("GET")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(SemanticAttributes.NET_PEER_NAME, httpHost.getHostName()),
+                            equalTo(SemanticAttributes.NET_PEER_PORT, httpHost.getPort()),
+                            equalTo(SemanticAttributes.HTTP_METHOD, "GET"),
+                            equalTo(AttributeKey.stringKey("net.protocol.name"), "http"),
+                            equalTo(AttributeKey.stringKey("net.protocol.version"), "1.1"),
+                            equalTo(SemanticAttributes.HTTP_URL, httpHost.toURI() + "/"),
+                            equalTo(SemanticAttributes.HTTP_STATUS_CODE, 200L),
+                            equalTo(SemanticAttributes.USER_AGENT_ORIGINAL, userAgent()),
+                            satisfies(
+                                SemanticAttributes.HTTP_RESPONSE_CONTENT_LENGTH,
+                                AbstractLongAssert::isPositive))));
+  }
+
+  @Test
+  // ignore deprecation interface
+  public void elasticsearchIndex() throws IOException {
+    client.index(
+        r ->
+            r.id("test-id")
+                .index("test-index")
+                .document(new Person("person-name"))
+                .timeout(t -> t.time("1s")));
```

Comment:
Usually tests are run in parallel which generates quite a bit of load. I would advise against using short timeouts like `1s` as this will most likely make your test flaky.